functions/api/inventory.js

```diff
@@ -0,0 +1,180 @@
+    var branchId = request.params.id
+    logger.info(`Retrieving inventory for a branch from firestore`)
+    const branchRef = db.collection(constants.BRANCHES).doc(branchId);
+    const branchSnapshot = await branchRef.get()
+    if (!branchSnapshot.exists) {
+        const error = new Error(`Requested Branch is not present in Firestore.`)
+        error.statusCode = 404
+        next(error)
+        return;
+    }
+    const inventory = await getInventory(branchRef, branchId, false)
+
+    logger.debug(`Returning inventory for branch ${branchSnapshot.data().name} to client.`);
+    response.status(200).send(inventory);
+});
+
+/**
+ * @description Route to retrieve inventory of products below threshold for a branch from firestore
+ * @returns Json object containing inventory of products below threshold for a branch
+ * @throws 400 if the branch does not exists in firestore
+ */
+router.get('/belowthreshold/:id', async (request, response, next) => {
+    var branchId = request.params.id
+    logger.info(`Retrieving inventory for a branch from firestore`)
+
+    const branchRef = db.collection(constants.BRANCHES).doc(branchId);
+    const branchSnapshot = await branchRef.get()
+    if (!branchSnapshot.exists) {
+        const error = new Error(`Requested branch is not present.`)
+        error.statusCode = 404
+        next(error)
+        return;
+    }
+    const inventory = await getInventory(branchRef, branchId, true)
+
+    logger.debug(`Returning inventory for branch ${branchSnapshot.data().name} to client.`);
+    response.status(200).send(inventory);
+});
+
+/**
+ * @description Route to retrieve inventory for all branches from firestore
+ * @returns Json object containing inventory for all branches
+ */
+router.get('/', async (request, response, next) => {
+    const inventories = {
+        "inventories": []
+    }
+    
+    const branchCollectionRef = db.collection(constants.BRANCHES)
+    const branchDocuments = await branchCollectionRef.listDocuments();
+    logger.info(`Retrieving inventory for  ${branchDocuments.length} branches from firestore`)
+    const inventoryArray = []
+    for(const doc of branchDocuments) {
+        const p1 = getInventory(branchCollectionRef.doc(doc.id), doc.id ,false)
+        inventoryArray.push(p1)
+    }
+    Promise.all(inventoryArray).then(inventory => {
+        inventories.inventories.push(inventory)
+        inventories[constants.TOTAL_BRANCHES] = branchDocuments.length
+        logger.debug(`Returning inventory for all products for all branches.`);
+        return response.status(200).send(inventories);
+    }).catch(err => {
+        err.statusCode = 500
+        next(err)
+        return;
+    })
+});
+
+/**
+ * @description Route to retrieve inventory for all products below threshold for all branches
+ * @returns Json object containing inventory for all branches
+ */
+router.get('/all/branches/belowthreshold/', async (request, response, next) => {
```

Comment:
https://github.com/Parikrama-inventory/backend/pull/9/files#diff-b94255622c82c86f6dbcc65ca661a1a3R41
can we have it consistent? /inventory/:branchId/belowThreshold - for individual branch and 
/inventory/branches/belowThreshold - for all branches

Reply:
I did the first change /inventory/:branchId/belowThreshold. 
For second change /inventory/branches/belowThreshold, it takes "branches" as the branchId for the previous route. So, I have kept it as /inventory/all/branches/belowThreshold